.github/workflows/maven.yml

```diff
@@ -29,4 +29,4 @@ jobs:
       - name: Test with Maven
         env:
           MAVEN_OPTS: "-Xms3g -Xmx5g -Djdk.xml.xpathExprGrpLimit=500 -Djdk.xml.xpathExprOpLimit=500"
-        run: mvn install  -B -V -Pminimal-fix-latest -D"java.util.logging.config.file"="logging.properties" -D"http.keepAlive"="false" -D"maven.wagon.http.pool"="false" -D"maven.wagon.httpconnectionManager.ttlSeconds"="120"
+        run: mvn test -B -V -Pminimal-fix-latest -D"java.util.logging.config.file"="logging.properties" -D"http.keepAlive"="false" -D"maven.wagon.http.pool"="false" -D"maven.wagon.httpconnectionManager.ttlSeconds"="120"
```

Comment:
`Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:2.8:unpack (unpack-dependencies) on project quickfixj-messages-fixt11: Artifact has not been packaged yet. When used on reactor artifact, unpack should be executed after packaging: see MDEP-98`

Never seen that one before.

Reply:
OK I see it now, It's because of unpacking the QFJ orchestration to be used in code generation.

This takes us into a philosophical territory. I think it's better to use packaged artefacts than to copy from relative locations in a multi-module build. This approach here is also didactic as a QFJ user customising messages would import and unpack the dependency in just the same way ( without having to check out an build the whole QFJ project ). If we do things the same way as we may expect a QFJ user to do them, this should help with maintainability.

If it is OK with you I think it's good to use mvn install, there are not any negative side effects I can think of as the build starts with a fresh local maven repo.

```
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-dependency-plugin</artifactId>
				<executions>
					<execution>
						<id>unpack-dependencies</id>
						<phase>initialize</phase>
						<goals>
							<goal>unpack</goal>
						</goals>
						<configuration>
							<artifactItems>
								<artifactItem>
									<groupId>org.quickfixj</groupId>
									<artifactId>quickfixj-orchestration</artifactId>
									<version>${project.version}</version>
									<includes>**/FIX*.xml,**/Orchestra*.xml</includes>
									<outputDirectory>${project.build.directory}/dependency/repository</outputDirectory>
								</artifactItem>
							</artifactItems>
						</configuration>
					</execution>
				</executions>
			</plugin>
```

The following shows that the build starts with a fresh local maven repo, artefacts are not left lying around from earlier builds.

[INFO] --------------< org.quickfixj:quickfixj-messages-fixt11 >---------------
[INFO] Building QuickFIX/J Message classes for FIXT 1.1 3.0.0-SNAPSHOT   [8/25]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-antrun-plugin/1.8/maven-antrun-plugin-1.8.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-antrun-plugin/1.8/maven-antrun-plugin-1.8.pom (3.3 kB at 53 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugins/27/maven-plugins-27.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-plugins/27/maven-plugins-27.pom (11 kB at 232 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-antrun-plugin/1.8/maven-antrun-plugin-1.8.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/plugins/maven-antrun-plugin/1.8/maven-antrun-plugin-1.8.jar (36 kB at 692 kB/s)

Reply:
mvn install should run integration tests, have I changed a default somewhere ?https://stackoverflow.com/questions/1399240/how-do-i-get-my-maven-integration-tests-to-run

Reply:
base has : [INFO] Tests run: 153, Failures: 0, Errors: 0, Skipped: 0 
I think the 614 tests are the integration tests...

Reply:
@chrjohn that's more like it, only problem is, there are test failures now.
2023-03-21T14:36:09.5779520Z [INFO] 
2023-03-21T14:36:09.5781950Z [ERROR] Tests run: 1422, Failures: 2, Errors: 0, Skipped: 0

Reply:
Sure - is generated base class for message cracker wrong ?
Yes there is a lot of logging that we can do without once we are confident in the overall structure.

Reply:
@chrjohn 
One thing I have noticed is a large delay in processing compared to some other tests, perhaps this is related to why there is a Heartbeat.

INFO: load test: F:\dev\git\quickfixj\quickfixj-core\target\test-classes\quickfix\test\acceptance\definitions\server\fix50\2r_UnregisteredMsgType.def
Mar 21, 2023 2:12:01 PM quickfix.test.acceptance.PrintComment run
INFO: # Message type is not supported by the application
Mar 21, 2023 **2:12:01** PM quickfix.mina.acceptor.AcceptorIoHandler sessionCreated
INFO: MINA session created: local=/127.0.0.1:9887, class org.apache.mina.transport.socket.nio.NioSocketSession, remote=/127.0.0.1:64223
Mar 21, 2023 **2:12:07** PM quickfix.test.acceptance.AcceptanceTestSuite$AcceptanceTest run
SEVERE: test F:\dev\git\quickfixj\quickfixj-core\target\test-classes\quickfix\test\acceptance\definitions\server\fix50\2r_UnregisteredMsgType.def failed with message: wrong msg type expected:<[j]> but was:<[0]>
Mar 21, 2023 2:12:07 PM quickfix.test.acceptance.AcceptanceTestSuite$AcceptanceTest load
INFO: load test: F:\dev\git\quickfixj\quickfixj-core\target\test-classes\quickfix\test\acceptance\definitions\server\fix50\2t_FirstThreeFieldsOutOfOrder.def
Mar 21, 2023 2:12:07 PM quickfix.test.acceptance.PrintComment run

Reply:
The delay is because the expected message has not been received. Then the connection will be open for some seconds and heartbeats are sent/ received. Eventually there will be a timeout.

Reply:
> The delay is because the expected message has not been received. Then the connection will be open for some seconds and heartbeats are sent/ received. Eventually there will be a timeout.

Is it received but an exception swallowed ? Other FIX50 cases seem to receive messages...

Reply:
Sounds like the best approach , I already spent 40 minutes gazing and guessing about the code ;-)

Reply:
Hi @david-gibbs-ig , I have added some temporary logging into `quickfix.MessageCracker` and it indeed looks like the thrown Exception is swallowed. But don't know where ATM.

https://github.com/david-gibbs-ig/quickfixj/blob/3a4d5c2bb982be46441fe540dfeb536543bfd2b3/quickfixj-core/src/main/java/quickfix/MessageCracker.java#L167